gui/mainwindow.cpp

```diff
@@ -59,6 +61,13 @@ MainWindow::MainWindow(TranslationHandler* th, QSettings* settings) :
     mCppStandardActions(new QActionGroup(this)),
     mExiting(false)
 {
+    QStringList CompletionList ;
+    CompletionList << "pointer dereference" << "missed break statement" << "syntax error" << "scope reducing"
+                   << "uninitialized variable" << "value never used"
+                   << "scanf" << "return" << "missed break" << "memory leak"<< "divison by zero" << "out of range" << "void *" ;
+    QCompleter *Complet = new QCompleter(CompletionList ,this) ;
```

Comment:
No leak as far as I see. Passing "this" to Qt class constructors normally means that the memory is deallocated when "this" is deallocated.